### PR TITLE
app-benchmarks/stress-ng: Bump to version 0.09.27

### PR DESCRIPTION
**Pull request by ConiKost**
Please see the Ticket, there is one QA Warning left..

Closes: https://bugs.gentoo.org/650140
Package-Manager: Portage-2.3.35, Repoman-2.3.9

### CODE REVIEW

File: app-benchmarks/stress-ng/files/makefile.patch [hunk 1]
```diff
@@ -0,0 +1,75 @@
+--- a/Makefile	2018-04-27 17:02:16.000000000 +0200
++++ b/Makefile	2018-05-05 00:15:00.687399505 +0200
+@@ -21,7 +21,7 @@
+ # Codename "portable pressure producer"
+ #
+ 
+-CFLAGS += -Wall -Wextra -DVERSION='"$(VERSION)"' -O2 -std=gnu99
++CFLAGS += -DVERSION='"$(VERSION)"'
```
Comment (mgorny) [inline app-benchmarks/stress-ng/files/makefile.patch]: `-W` flags are fine, actually (except for `-Werror`).

File: app-benchmarks/stress-ng/files/makefile.patch [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ 
+ #
+ # Pedantic flags
+@@ -314,12 +314,10 @@
+ .o: stress-ng.h Makefile
+ 
+ .c.o: stress-ng.h Makefile $(SRC)
+-	@echo "CC $<"
+-	@$(CC) $(CFLAGS) -c -o $@ $<
++	$(CC) $(CFLAGS) $(LDFLAGS) -c -o $@ $<
```
Comment (mgorny) [inline app-benchmarks/stress-ng/files/makefile.patch]: If it's rule for compiling object files, LDFLAGS are unnecessary. I suppose you may also remove it altogether and let make do its default (if it works).

File: app-benchmarks/stress-ng/files/makefile.patch [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ 
+ stress-ng: info $(OBJS)
+-	@echo "LD $@"
+-	@$(CC) $(CPPFLAGS) $(CFLAGS) $(OBJS) -lm $(LDFLAGS) -lc -o $@
++	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $(OBJS) -lm $(LDFLAGS) -lc -o $@
```
Comment (mgorny) [inline app-benchmarks/stress-ng/files/makefile.patch]: You have LDFLAGS twice. `-lc` seems unnecessary.

File: app-benchmarks/stress-ng/files/makefile.patch [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+@@ -398,10 +394,8 @@
+ 	STRESS_NG=./stress-ng debian/tests/fast-test-all
+ 
+ .PHONY: install
+-install: stress-ng stress-ng.1.gz
++install: stress-ng
+ 	mkdir -p ${DESTDIR}${BINDIR}
+ 	cp stress-ng ${DESTDIR}${BINDIR}
+-	mkdir -p ${DESTDIR}${MANDIR}
+-	cp stress-ng.1.gz ${DESTDIR}${MANDIR}
```
Comment (mgorny) [inline app-benchmarks/stress-ng/files/makefile.patch]: Why not fix the rule?

Comment (ConiKost) [reply app-benchmarks/stress-ng/files/makefile.patch]: You mean uncompress man, and remove here the *.gz?

File: app-benchmarks/stress-ng/files/makefile.patch (hunk 3 above)
Comment (ConiKost) [reply app-benchmarks/stress-ng/files/makefile.patch]: Whoops, seems I didn't saw that. Thanks.

File: app-benchmarks/stress-ng/files/makefile.patch (hunk 1 above)
Comment (ConiKost) [reply app-benchmarks/stress-ng/files/makefile.patch]: Good to know, thanks.

File: app-benchmarks/stress-ng/files/makefile.patch (hunk 2 above)
Comment (ConiKost) [reply app-benchmarks/stress-ng/files/makefile.patch]: Seems to work, when I do remove it. Pushed now a new eBuild. I also set QA_WX_LOAD. Please see, if I should better remove it.

File: app-benchmarks/stress-ng/files/makefile.patch [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+ 	@sync
+ 
+ .PHONY: info
+@@ -339,8 +336,7 @@
+ 		sed '$$ s/.$$//' >> apparmor-data.c
+ 	@echo "};" >> apparmor-data.c
+ 	@echo "const size_t g_apparmor_data_len = sizeof(g_apparmor_data);" >> apparmor-data.c
+-	@echo "CC $<"
+-	@$(CC) -c apparmor-data.c -o apparmor-data.o
++	$(CC) -c apparmor-data.c -o apparmor-data.o
```
Comment (mgorny) [inline app-benchmarks/stress-ng/files/makefile.patch]: Missing CPPFLAGS, CFLAGS.

File: app-benchmarks/stress-ng/files/makefile.patch [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+ #
+@@ -353,20 +349,15 @@
+ stress-personality.c: personality.h
+ 
+ perf.o: perf.c perf-event.c
+-	@$(CC) -E perf-event.c | grep "PERF_COUNT" | sed 's/,/ /' | awk {'print "#define _SNG_" $$1 " (1)"'} > perf-event.h
+-	@echo CC $<
+-	@$(CC) $(CFLAGS) -c -o $@ $<
++	$(CC) -E perf-event.c | grep "PERF_COUNT" | sed 's/,/ /' | awk {'print "#define _SNG_" $$1 " (1)"'} > perf-event.h
++	$(CC) $(CFLAGS) -c -o $@ $<
```
Comment (mgorny) [inline app-benchmarks/stress-ng/files/makefile.patch]: Missing CPPFLAGS.

File: app-benchmarks/stress-ng/files/makefile.patch [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+ 
+ stress-vecmath.o: stress-vecmath.c
+-	@echo CC $<
+-	@$(CC) $(CFLAGS) -fno-builtin -c -o $@ $<
++	$(CC) $(CFLAGS) -fno-builtin -c -o $@ $<
```
Comment (mgorny) [inline app-benchmarks/stress-ng/files/makefile.patch]: Likewise.

File: app-benchmarks/stress-ng/files/makefile.patch (hunk 5 above)
Comment (ConiKost) [reply app-benchmarks/stress-ng/files/makefile.patch]: Whoops, fixed this and rest. Could you tell me, why CPPFLAGS? Should I always add them?

**Comment by mgorny** [comment]
Merging, thanks!

**Pull request closed without merging**